### ISSUE TITLE
PERMISSION_DENIED using firestore emulator and Java SDK

### ISSUE THREAD

**Issue by ddadon10** [firebase/firebase-tools#3380]
The Java SDK does not bypass the emulator security rules.
This seems to be similar to #3258, so that's why I am raising it here.

**Platform: macOS 10.15.7 - Java 8**

### [REQUIRED] Steps to reproduce
1) Set the following environment variable:
```bash
export FIRESTORE_EMULATOR_HOST="localhost:8080"
```

2) Set the following security rules for the firestore emulator
```
rules_version = '2';
service cloud.firestore {
  match /databases/{database}/documents {
    match /organizations/{organization} {
      allow read: if false;
    }
  }
}
```
3) Launch the firestore emulator using the following command: `firebase emulators:start --project=test-local-project`

4) Create a collection `organizations` and an document `acme` in the firestore emulator

5) In your `pom.xml`, add the following dependency
```xml
    <dependency>
      <groupId>com.google.cloud</groupId>
      <artifactId>google-cloud-firestore</artifactId>
      <version>2.3.0</version>
    </dependency>
```

6) Add this file to your java project
```java
package org.example;

import com.google.cloud.firestore.DocumentSnapshot;
import com.google.cloud.firestore.Firestore;
import com.google.cloud.firestore.FirestoreOptions;
import java.util.concurrent.ExecutionException;

public class App {

  public static void main(String[] args) throws ExecutionException, InterruptedException {

    Firestore firestore = FirestoreOptions.newBuilder()
        .setHost("localhost:8080")
        .setProjectId("test-local-project")
        .build()
        .getService();
    DocumentSnapshot snap = firestore.collection("organizations")
        .document("acme")
        .get().get();
  }
}

```

### [REQUIRED] Expected behavior
The Server SDK should bypass the security rules, as mentioned in [Getting started with security rules](https://cloud.google.com/firestore/docs/security/get-started), and the program should exit correctly without error.

### [REQUIRED] Actual behavior
The following error is thrown: ` io.grpc.StatusRuntimeException: PERMISSION_DENIED`

If I change the security rules to `allow read: if true;`, it works.

**Comment by samtstern** [comment]
@ddadon10 rather than using `.setHost("localhost:8080")` could you try setting the `FIRESTORE_EMULATOR_HOST` environment variable to `localhost:8080` on your machine?

**Comment by ddadon10** [comment]
Hi @samtstern, thanks for your fast answer.

I already had this environment variable set when I opened this issue.

Though, removing `.setHost("localhost:8080")` solved the issue.

I got fooled by `firestoreOptions.host`, which is `firestore.googleapis.com:443` even when the emulator is running, but ignored because there is also a `firestoreOptions.emulatorHost` field (which is automatically set the `localhost:8080` in my case because of the env variable).

Thanks for your help and happy I/O!
